runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPTransformMatchers.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.dataflow;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayDeque;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.runners.PTransformMatcher;
+import org.apache.beam.sdk.runners.TransformHierarchy;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.PTransform;
+
+/**
+ * A set of {@link PTransformMatcher PTransformMatchers} that are used in the Dataflow Runner and
+ * not general enough to be shared between runners.
+ */
+class DataflowPTransformMatchers {
+  private DataflowPTransformMatchers() {}
+
+  /**
+   * Returns a {@link PTransformMatcher} that matches {@link PTransform}s of class {@link
+   * Combine.GroupedValues} that will be translated into CombineValues transforms in Dataflow's Job
+   * API and skips those that should be expanded into ParDos.
+   *
+   * @param fnApiEnabled Flag indicating whether this matcher is being retrieved for a fnapi or
+   *     non-fnapi pipeline.
+   */
+  static PTransformMatcher combineValuesTranslation(boolean fnApiEnabled) {
```

Comment:
I would prefer to avoid using static methods. How about this:
1. rename class to DataflowPTransformMatchersProvider
2. Rename combineValuesTranslation to get(boolean isFnApiEnabled)
3. in DataflowRunner create instance of this class, or better utilize dependency injection and pass it to DataflowRunner constructor at [DataflowRunner.java#L330](https://github.com/apache/beam/blob/0ba9168cf087df14e635fbf360f6615863e6ea2c/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L330)

It is better to do first two steps anyway. 3rd step is good for future code testability and cleaner communication of dependencies.

Reply:
Lets move this method to DataflowRunner so that DataflowPTransformMatchers plays intended role of namespace for Matchers.